analyses/molecular-subtyping-EWS/01-reclassify_as_ewings.Rmd

```diff
@@ -0,0 +1,64 @@
+---
+title: "Reclassify integrated_diagnosis,short_histology metadata column for samples that have hallmark Ewing Sarcoma fusions"
+output: html_notebook
+author: K S Gaonkar for D3b
+date: January 2020
+---
+
+Identify "integrated_diagnosis","short_histology" from clinical file for sample IDs with hallmark EWSR1 fusions and reclassify as Ewing Sarcoma (integrated_diagnosis) and EWS (short_histology)
+
+```{r}
+library("tidyverse")
+```
+
+
+### Read in data
+
+```{r}
+# to get root directory
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+# data directory
+dataDir <- file.path(root_dir, "data")
+# histology files
+clinical <- readr::read_tsv(file.path(dataDir, "pbta-histologies.tsv")) %>%
+  select("Kids_First_Biospecimen_ID","integrated_diagnosis","short_histology","broad_histology","Kids_First_Participant_ID","experimental_strategy")
+# get ewings fusion calls from fusion-summary results file
+ewings_foi_bsids <- readr::read_tsv(file.path(root_dir,"analyses","fusion-summary","results","fusion_summary_ewings_foi.tsv")) 
+
+# get subset folder
+results_dir<-"results"
+
+# create if doesn't exist
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+
+
+```
+
+### reclassify integrated diagnosis and short_histology of samples that have hallmark ewings sarcoma EWSR1 fusion
+```{r}
+reclassifySamplesEwings<-ewings_foi_bsids %>% 
+  dplyr::mutate(integrated_diagnosis_reclassified=dplyr::case_when(.data$`EWSR1--FLI1` !=0 | .data$`EWSR1--PATZ1` !=0 | .data$`EWSR1--ERG` !=0  |.data$`FUS--ERG` !=0  ~ "Ewing Sarcoma"),
+                short_histology_reclassified=dplyr::case_when(.data$`EWSR1--FLI1` !=0 | .data$`EWSR1--PATZ1` !=0 | .data$`EWSR1--ERG` !=0  |.data$`FUS--ERG` !=0 ~ "EWS"),
+                broad_histology_reclassified=dplyr::case_when(.data$`EWSR1--FLI1` !=0 | .data$`EWSR1--PATZ1` !=0 | .data$`EWSR1--ERG` !=0  |.data$`FUS--ERG` !=0 ~ "Mesenchymal non-meningothelial tumor"))
+
+clinical<-clinical %>% left_join(reclassifySamplesEwings ,by=c("Kids_First_Biospecimen_ID"))
+
+readr::write_tsv(clinical,file.path(results_dir,"reclassified_ewings_samples.tsv"))
```

Comment:
I apologize for missing this the first time around, but I've thought about this table and how what you're providing aligns with what is in the other subtyping modules. 

With the way the table that is saved as `reclassified_ewings_samples.tsv` is currently constructed, I believe WGS biospecimen samples from the same sample would be `NA` when you'd likely also want to reclassify those.

In essence, I think we want a table contains

| participant ID | sample ID | RNA-seq BSID | DNA-seq BSID | ... here are the disease type columns .. |
|--------------|------------|----------------|---------------|----------------------------------------|


What I've put together below achieves that:

```R
# get a vector of biospecimen IDs for those biospecimens that should be reclassified
have_ews_fusion <- rowSums(ewings_foi_bsids[, -1]) > 0
ews_rnaseq_bsids <- ewings_foi_bsids %>%
  filter(have_ews_fusion) %>%
  pull(Kids_First_Biospecimen_ID)

# recode the different disease labels for RNA-seq samples only
# because we have fusion information for RNA-seq samples
clinical_rnaseq <- clinical %>%
  filter(experimental_strategy == "RNA-Seq") %>%
  select(-experimental_strategy) %>%
  # the `TRUE ~` steps leave the original disease label 
  mutate(
    integrated_diagnosis_reclassified = case_when(
      Kids_First_Biospecimen_ID %in% ews_rnaseq_bsids ~ "Ewing Sarcoma",
      TRUE ~ integrated_diagnosis
    ),
    short_histology_reclassified = case_when(
      Kids_First_Biospecimen_ID %in% ews_rnaseq_bsids ~ "EWS",
      TRUE ~ short_histology
    ),
    broad_histology_reclassified = case_when(
      Kids_First_Biospecimen_ID %in% ews_rnaseq_bsids ~ "Mesenchymal non-meningothelial tumor",
      TRUE ~ broad_histology
    )
  ) %>%
  # remove this step if you want to filter later to show what has changed
  select(-integrated_diagnosis, 
         -short_histology, 
         -broad_histology)

# we're going to join the WGS samples to RNA-seq
# so that the WGS biospecimens that correspond to RNA-seq specimens 
# are reclassified as well
clinical_wgs <- clinical %>%
  filter(experimental_strategy == "WGS") %>%
  select(Kids_First_Biospecimen_ID, 
         Kids_First_Participant_ID, 
         sample_id)

# this drops normal samples
clinical <- clinical_rnaseq %>% 
  inner_join(clinical_wgs,
             by = c("Kids_First_Participant_ID",
                    "sample_id"),
             suffix = c("_RNA", "_DNA")) %>%
  select(Kids_First_Participant_ID,
         Kids_First_Biospecimen_ID_RNA,
         Kids_First_Biospecimen_ID_DNA,
         sample_id,
         everything())
```

Reply:
This relies on the suggested change that includes `sample_id`.

Reply:
Thank you for the snippet that fixes the WGS NA issue. I think we need to add sample_type as well to only annotated Tumor samples as EWS right? https://github.com/kgaonkar6/OpenPBTA-analysis/blob/9cc033d0bfc421ae0c22701ff18ed5dac78e5c20/analyses/molecular-subtyping-EWS/01-reclassify_as_ewings.Rmd#L77

Reply:
Because you're joining to the RNA-seq data by participant ID and sample ID, I _believe_ it is equivalent to filtering to tumor (there are no normal RNA-seq samples).

Reply:
Oh I meant while creating clinical_wgs; we would need to remove the WGS normal from here ?

Reply:
Ah, my point was that because of the way we are joining to the RNA-seq data and that's the only use of `clinical_wgs` - the end result would be the same regardless of filtering to tumors or not (untested).

Reply:
Nope, not equivalent - I made an assumption that the `sample_id` would be different between tumor and normal, but I guess it makes sense that they are the same if they are an event ID (https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/260#issuecomment-552929142). Good call on the filtering!

Reply:
With the inner join, we're basically saying we're unable to do this reclassification in the absence of RNA-seq (fusion calls to be exact), which I think is what we want. Do you agree @kgaonkar6 ?

Reply:
yes that's right!